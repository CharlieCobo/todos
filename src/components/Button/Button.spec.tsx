import { describe, it, expect, vi, afterEach } from 'vitest';
import { render, fireEvent, screen, cleanup } from '@testing-library/react';

import { Button, Props } from '.';

describe('<Button />', () => {
  afterEach(cleanup);

  const initialProps: Props = {
    label: 'Click Me!',
    onClick: vi.fn(),
  };

  it('Should match with the snapshot', () => {
    const { container } = render(<Button {...initialProps} />);

    expect(container).toMatchSnapshot();
  });

  it('Should exec the onClick function when user hit the button', () => {
    render(<Button {...initialProps} />);

    const btn = screen.getByRole('button');

    fireEvent.click(btn);

    expect(initialProps.onClick).toHaveBeenCalled();
  });
});
