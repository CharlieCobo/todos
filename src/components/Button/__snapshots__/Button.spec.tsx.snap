// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Button /> > Should match with the snapshot 1`] = `
<div>
  <button
    class="css-1bi00ci"
  >
    Click Me!
  </button>
</div>
`;
